### PR TITLE
#152799661 create tests for the user sign in and sign up API routes

### PR DESCRIPTION
**Pull request by chinemelu**
#### What does this PR do?
create tests for the user sign up  and the user sign in API routes.

#### Description of Task to be completed?
install and setup the following dependencies
chai v4.1.2
chai-http v3.0.0
mocha  v4.0.1

write tests for the following API routes
POST: /api/v1/users
POST: /api/users/login

#### What are the relevant pivotal tracker stories?
#152799661

### CODE REVIEW

**Comment by chinemelu** [comment]
@andela-uibrahim, please review this PR.

File: server/test/test.js [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+import db from '../models/index';
+
+
+chai.use(chaihttp);
+chai.should();
+const { expect } = chai;
+
+describe('Users', () => {
+  describe('POST: /api/v1/users/', () => {
+    beforeEach((done) => {
```
Comment (freemile2ru) [inline server/test/test.js:13]: why are you creating this user before each?

File: server/test/test.js [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+      const user = {
+        username: null,
+        email: 'testemail@test.com',
+        isAdmin: false,
+        password: '@testPassword1',
+        reEnterPassword: '@testPassword1'
+      };
+      chai.request(server)
+        .post('/api/v1/users/')
+        .set({ token: process.env.TEST_TOKEN })
```
Comment (freemile2ru) [inline server/test/test.js:40]: why is it necessary to have a token before registering?

File: server/test/test.js [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+          res.body.errors[0].should.eql('Username field must not be empty.');
+          res.body.errors[1].should.eql('Username must be between 4-15 characters \n' +
+          'long, it must include at least one lowercase alphabet, and it must consist of \n' +
+          'only underscores,lowercase alphabets, numbers,\n' +
+          'and no spaces in between characters.');
+          done();
+        });
+    });
+    
+    it('it should not create a user if the username is fewer than 4 characters', (done) => {
```
Comment (freemile2ru) [inline server/test/test.js:56]: I don't think this is necessary

File: server/test/test.js [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
+          'only underscores,lowercase alphabets, numbers,\n' +
+          'and no spaces in between characters.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the username contains an uppercase \n' +
+    'character', (done) => {
+      const user = {
+        username: 'Test',
```
Comment (freemile2ru) [inline server/test/test.js:135]: I don't think this is necessary you can converted input to lowercase at the backend before saving

File: server/test/test.js [hunk 5]
```diff
@@ -0,0 +1,665 @@
...
+          res.body.errors[0].should.eql('The email you entered is invalid, please try again.')
+          done();
+        });
+    });
+
+    it('it should not create a user if there are any uppercase characters \n' +
+       'contained in the email', (done) => {
+      const user = {
+        username: 'test',
+				email: 'Testemail@test.com',
```
Comment (freemile2ru) [inline server/test/test.js:232]: this again can be handled on the backend

File: server/test/test.js [hunk 6]
```diff
@@ -0,0 +1,665 @@
...
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Password must be between 8-100 characters long,\n' +
+            'and it must include one lowercase character, one uppercase character,\n' +
+            'a number, and a special character.');
+          res.body.errors[1].should.eql('Passwords do not match, please try again.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the password has no lowercase characters', (done) => {
```
Comment (freemile2ru) [inline server/test/test.js:326]: why?

File: server/test/test.js [hunk 7]
```diff
@@ -0,0 +1,665 @@
...
+          res.body.should.have.property('errors');
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Password must be between 8-100 characters long,\n' +
+            'and it must include one lowercase character, one uppercase character,\n' +
+            'a number, and a special character.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the password has no uppercase characters', (done) => {
```
Comment (freemile2ru) [inline server/test/test.js:350]: why?

File: server/test/test.js [hunk 8]
```diff
@@ -0,0 +1,665 @@
...
+          res.body.should.have.property('errors');
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Password must be between 8-100 characters long,\n' +
+            'and it must include one lowercase character, one uppercase character,\n' +
+            'a number, and a special character.');
+          done();
+        });
+    });
+
+    it('it should not create a user if the password does not contain a number', (done) => {
```
Comment (freemile2ru) [inline server/test/test.js:374]: why?

File: server/test/test.js [hunk 9]
```diff
@@ -0,0 +1,665 @@
...
+          res.body.should.have.property('errors');
+          res.body.errors.should.be.a('array');
+          res.body.errors[0].should.eql('Password must be between 8-100 characters long,\n' +
+            'and it must include one lowercase character, one uppercase character,\n' +
+            'a number, and a special character.');
+          done();
+        });
+    });
+  
+    it('it should not create a user if the password has no special character', (done) => {
```
Comment (freemile2ru) [inline server/test/test.js:398]: hahahah why naaaa

**Pull request merged**